clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -547,23 +549,33 @@ object GarbageCollector {
   private def getPreviousRunID(
       incrementalRun: Boolean,
       storageNS: String,
-      configMapper: ConfigMapper
+      configMapper: ConfigMapper,
+      iterations: Int
   ): String = {
     if (!incrementalRun) {
       return ""
     }
+    if (iterations < 1) {
+      throw RunIDException("Run ID iteration number cannot be smaller than 1")
+    }
     var previousRunID = ""
     val runIDsPath = new Path(s"${storageNS.stripSuffix("/")}/_lakefs/retention/gc/run_ids/")
     val runIDsFS = runIDsPath.getFileSystem(configMapper.configuration)
     val runIDs = runIDsFS.listFiles(runIDsPath, false)
-    if (runIDs.hasNext) {
-      previousRunID = runIDs.next().getPath.getName
-      println(s"----------------------- Using previous RUN ID: $previousRunID")
-    } else {
+    if (!runIDs.hasNext) {
       runIDsFS.close()
       throw RunIDException("No previous run ID")
     }
+    var runIDObject: LocatedFileStatus = null
+    for(_ <- 0 until iterations) {
```

Comment:
This is a bit problematic in production: when we _start_ a new user, the first "iterations" runs cannot be incremental.  I suggest that if there are _not enough_ previous runs, we simply run non-incrementally.

My use-case: I want to run incremental GC on lakeFS.Cloud with iterations set to a constant (say)  "10".  This proposal will let a single configuration run cleanly.  Think of it as "N iterations back if possible, otherwise run non-incrementally".